### ISSUE TITLE
Difference between addJob and updateJob?

### ISSUE THREAD

**Issue by Caerbannog** [doowb/firebase-cron#7]
There is currently no difference between addJob and updateJob, see:
https://github.com/doowb/firebase-cron/blob/master/index.js#L65
https://github.com/doowb/firebase-cron/blob/master/index.js#L86

I would have expected updateJob to execute the job if it is pending before recomputing _getNextDateFrom. Currently it does not, is this intentional?

Scenario where it would be useful:

* daily job A is created by these commands:
    ```
    cron = new Cron()
    cron.updateJob(job A)
    ```
* the server is stopped
* the nextRun date for job A (00h00) passes
* the server is restarted and executes the same commands as in the first step
* the job A is executed by updateJob() because the 00h00 deadline was missed
* updateJob updates nextRun and we're back to normal without missing a task

**Comment by doowb** [comment]
You're correct that there are no differences between `addJob` and `updateJob`. I had stubbed out the methods for the api before implementing them and didn't really see how they could differ.

I like your suggestion of not overwriting the current record, but I'm not sure if executing the job first, then updating is the correct approach.

I usually have 2 scripts, one for initializing my jobs (if they aren't added dynamically) and another for starting the cron server that executes the jobs. This lets me start the cron server and it will pick up any jobs that were missed while the server was stopped.

I think what you can do is check to see which jobs have already been registered, then register any that are missing (this should provide the affect your looking for).

 ```js
var each = require('async-each');
var jobs = {
  'job-1': {
    pattern: '00 00 00 * * *',
    data: {foo: 'bar'}
  },
  'job-2': {
    pattern: '00 00 02 * * *',
    data: {baz: 'qux'}
  }
};

cron.getJobs(function(err, existing) {
  if (err) return console.error(err);
  each(Object.keys(jobs), function(name, next) {
    if (existing.hasOwnProperty(name)) {
      return next();
    }
    var job = jobs[name];
    cron.addJob(name, job.pattern, job.data, next);
  }, function(err) {
    if (err) return console.error(err);
    console.log('finished adding jobs');
  });
});
```

I'm adding this here just as an example and show anyone else seeing this might find it useful.

I think more options or other api methods could be useful for handling some of these scenarios. Like a flag to specify that `addJob` should only add the job if it doesn't already exist. Or maybe that should be the default and `updateJob` should always add/update a job.
